analysis/pseudobulk-bulk-prediction/scripts/sync-data-files.R

```diff
@@ -0,0 +1,137 @@
+# 1. First, we identify samples of interest as non-multiplexed samples from solid tumors with paired bulk data
+# 2. Second, we sync all associated bulk `quant.sf` files and single-cell `_processed.rds` files from S3 needed for analysis
+# All files are organized in `<project id>/<sample id>/`
+
+
+renv::load()
+library(optparse)
+
+
+# Parse options --------
+option_list <- list(
+  make_option(
+    "--output_dir",
+    type = "character",
+    help = "Output directory to save synced data files organized by project/sample"
+  ),
+  make_option(
+    "--sample_metadata_file",
+    type = "character",
+    default = here::here("s3_files", "scpca-sample-metadata.tsv"),
+    help = "Path to ScPCA sample metadata file."
+  ),  
+  make_option(
+    "--library_metadata_file",
+    type = "character",
+    default = here::here("s3_files", "scpca-library-metadata.tsv"),
+    help = "Path to ScPCA library metadata file."
+  )
+)
+opts <- parse_args(OptionParser(option_list = option_list))
+
+# Define files and directories ---------
+
+stopifnot(
+  "Metadata files could not be found. Were they synced?" = all(file.exists(c(opts$library_metadata_file, opts$sample_metadata_file)))
+)
+s3_bulk_path <- "s3://nextflow-ccdl-results/scpca-prod/checkpoints/salmon" #/library_id/quant.sf
+s3_processed_path <- "s3://nextflow-ccdl-results/scpca-prod/results" #/project_id/sample_id/library_id_processed.rds
+
+fs::dir_create(opts$output_dir)
+
+
+## Determine samples of interest -----------
+
+library_metadata <- readr::read_tsv(opts$library_metadata_file, show_col_types = FALSE)
+sample_metadata <- readr::read_tsv(opts$sample_metadata_file, show_col_types = FALSE)
+
+# we can't include multiplexed in this analysis, so we'll find those for removal
+cellhash_samples <- library_metadata |>
+  dplyr::filter(stringr::str_detect(technology, "cellhash")) |>
+  dplyr::pull(scpca_sample_id) |>
+  stringr::str_split(pattern = ";") |>
+  purrr::reduce(union)
+
+# all possible samples of interest
+all_bulk_samples <- library_metadata |>
+  dplyr::filter(
+    !(scpca_sample_id %in% cellhash_samples), 
+    seq_unit == "bulk"
+  ) |>
+  dplyr::pull(scpca_sample_id) |>
+  unique()
+
+
+# keep only solid tumors with directly paired single-cell and bulk
+solid_bulk_samples <- sample_metadata |>
+  dplyr::filter(scpca_sample_id %in% all_bulk_samples) |>
+  dplyr::filter(
+    !stringr::str_detect(diagnosis, "leukemia"),
+    !stringr::str_detect(diagnosis, "Non-cancerous")
+  ) |>
+  dplyr::pull(scpca_sample_id)
+
+# Create data frame to iterate over for syncing single-cell files
+# we do this first since there are fewer single-cell than corresponding bulk
+sync_sc_df <- library_metadata |>
+  dplyr::filter(
+    scpca_sample_id %in% solid_bulk_samples,
+    seq_unit %in% c("cell", "nucleus")
+  ) |>
+  dplyr::mutate(
+    s3_dir = file.path(s3_processed_path, scpca_project_id, scpca_sample_id),
+    output_dir = file.path(opts$output_dir, scpca_project_id, scpca_sample_id)
+  ) |>
```

Comment:
because of how `aws sync` works, you don't really need either of these; you can just give it the base dir and let it find the correct library files based on the `include` pattern. But it isn't going to hurt either really, and it _is_ needed in the next one, so.